Software entries for NestML, NIR, and KNP

Jegp opened the pull request.

neural-loop committed:
feat: Add NESTML, NIR, and KNP content

Adds markdown files and logos for NESTML, NIR, and KNP.

neural-loop committed:
feat(related-content): Improve related content linking

neural-loop committed:
feat: nir: Update NIR page and workshop with new tags and links

neural-loop committed:
fix(nir): Corrected grammar in overview section

Jegp commented on .github/workflows/main.yml:
```diff
@@ -2,7 +2,7 @@ name: Deploy Hugo site to Pages
 on:
   push:
-    branches: ["main", "student-talk"]
+    branches: ["main", "software-entries"]
```
Same question [as the other PR](https://github.com/open-neuromorphic/open-neuromorphic.github.io/pull/285#discussion_r2202907725): do we want to publish the docs when the branch gets pushed?

Jegp commented on content/neuromorphic-computing/software/snn-frameworks/knp/index.md:
```diff
@@ -0,0 +1,27 @@
+---
+title: "Kaspersky Neuromorphic Platform"
```
Ok, I *never* heard about this before. Nice find!

Jegp commented on content/neuromorphic-computing/software/snn-frameworks/nestml/index.md:
```diff
@@ -0,0 +1,23 @@
+---
+title: "NESTML"
+type: neuromorphic-software
```
Judging by the description, isn't this more like a (intermediate) language similar to NIR? In that case, I would classify it as a data tool more than a simulator.

neural-loop replied:
Thanks - that makes sense. I'll move it to data tools

Jegp reviewed (commented):
All in all, nice work!

I have one small comment regarding the classification of NestML as a simulator. Otherwise, LGTM!

neural-loop committed:
Refactor: Move NestML documentation and assets

Body: Moved NestML documentation and related image to the `data-tools` directory. Updated workflow to reflect branch changes.
